app/src/sync/process-initialization.ts

```diff
@@ -44,23 +44,13 @@ import {ensure_project_databases} from './projects';
 
 // called on startup to get the initial set of projects
 export async function update_directory() {
-  let listings = await getAllListings();
+  const listings = await getAllListings();
```

Comment:
This code is non-obvious what's going on. From memory you are building out a local version of the directories database by fetching the details from the API? Would be good to have function doc string and inline comments here.